pkg/reconciler/buildrun/resources/params.go

```diff
@@ -350,7 +350,7 @@ func isStepReferencingParameter(step *pipeline.Step, paramName string) bool {
 		// Tekton reference: https://github.com/tektoncd/pipeline/blob/main/docs/tasks.md#using-variable-substitution
 		fmt.Sprintf("$(params.%s", paramName),
 		fmt.Sprintf("$(params['%s']", paramName),
-		fmt.Sprintf("$(params[\"%s\"]", paramName),
+		fmt.Sprintf("$(params[%q]", paramName),
```

Comment:
This is incorrect. We basically look for all possible things from: https://github.com/tektoncd/pipeline/blob/main/docs/tasks.md#using-variable-substitution. You can fix line 352 to use `%q`.